crates/math/src/uint.rs

```diff
@@ -506,29 +663,7 @@ impl Uint512 {
 
 #[cfg(test)]
 mod tests {
-    use {super::*, crate::Dec128, proptest::prelude::*};
-
-    /// Make sure we can't multiply a positive integer by a negative decimal.
-    #[test]
-    fn multiply_fraction_by_negative() {
-        let lhs = Uint128::new(123);
-
-        // Multiplying with a negative fraction should fail
-        let rhs = Dec128::from_str("-0.1").unwrap();
-        assert!(lhs.checked_mul_dec_floor(rhs).is_err());
-        assert!(lhs.checked_mul_dec_ceil(rhs).is_err());
-        assert!(lhs.checked_div_dec_floor(rhs).is_err());
-        assert!(lhs.checked_div_dec_ceil(rhs).is_err());
-
-        // Multiplying with negative zero is allowed though
-        let rhs = Dec128::from_str("-0").unwrap();
-        assert!(lhs.checked_mul_dec_floor(rhs).unwrap().is_zero());
-        assert!(lhs.checked_mul_dec_ceil(rhs).unwrap().is_zero());
-
-        // Dividing by zero should fail
-        assert!(lhs.checked_div_dec_floor(rhs).is_err());
-        assert!(lhs.checked_div_dec_ceil(rhs).is_err());
-    }
```

Comment:
TODO: this test relies on signed numbers which I have deleted for now. Add back once have signed numbers. @Rhaki